src/query/union/mod.rs

```diff
@@ -0,0 +1,304 @@
+mod bitset_union;
+mod buffered_union;
+mod simple_union;
+
+pub use bitset_union::BitSetPostingUnion;
+pub use buffered_union::BufferedUnionScorer;
+pub use simple_union::SimpleUnion;
+
+#[cfg(test)]
+mod tests {
+
+    use std::collections::BTreeSet;
+
+    use common::BitSet;
+
+    use super::{SimpleUnion, *};
+    use crate::docset::{DocSet, TERMINATED};
+    use crate::postings::tests::test_skip_against_unoptimized;
+    use crate::query::score_combiner::DoNothingCombiner;
+    use crate::query::union::bitset_union::BitSetPostingUnion;
+    use crate::query::{BitSetDocSet, ConstScorer, VecDocSet};
+    use crate::{tests, DocId};
+
+    pub fn vec_doc_set_from_docs_list(
+        docs_list: &[Vec<DocId>],
+    ) -> impl Iterator<Item = VecDocSet> + '_ {
+        docs_list.iter().cloned().map(VecDocSet::from)
+    }
+    pub fn union_from_docs_list(docs_list: &[Vec<DocId>]) -> Box<dyn DocSet> {
+        Box::new(BufferedUnionScorer::build(
+            vec_doc_set_from_docs_list(docs_list)
+                .map(|docset| ConstScorer::new(docset, 1.0))
+                .collect::<Vec<ConstScorer<VecDocSet>>>(),
+            DoNothingCombiner::default,
+        ))
+    }
+
+    fn posting_list_union_from_docs_list(docs_list: &[Vec<DocId>]) -> Box<dyn DocSet> {
+        Box::new(BitSetPostingUnion::build(
+            vec_doc_set_from_docs_list(docs_list).collect::<Vec<VecDocSet>>(),
+            bitset_from_docs_list(docs_list),
+        ))
+    }
+    fn simple_union_from_docs_list(docs_list: &[Vec<DocId>]) -> Box<dyn DocSet> {
+        Box::new(SimpleUnion::build(
+            vec_doc_set_from_docs_list(docs_list).collect::<Vec<VecDocSet>>(),
+        ))
+    }
+    fn bitset_from_docs_list(docs_list: &[Vec<DocId>]) -> BitSetDocSet {
+        let max_doc = docs_list
+            .iter()
+            .flat_map(|docs| docs.iter())
+            .cloned()
```

Comment:
```suggestion
            .flat_map(|docs| docs.iter().copied())
```

Reply:
Yes, they are sorted. It's test code, so I think the simplest version should be fine